Source/Core/VideoBackends/D3D12/D3DBase.cpp

```diff
@@ -0,0 +1,924 @@
+
+		CheckHR(factory->CreateSwapChain(command_queue, &swap_chain_desc, &s_swapchain));
+
+		s_current_back_buf = 0;
+
+		factory->Release();
+	}
+
+	if (FAILED(hr))
+	{
+		MessageBox(wnd, _T("Failed to initialize Direct3D.\nMake sure your video card supports Direct3D 12 and your drivers are up-to-date."), _T("Dolphin Direct3D 12 backend"), MB_OK | MB_ICONERROR);
+		SAFE_RELEASE(s_swapchain);
+		return E_FAIL;
+	}
+
+	ID3D12InfoQueue* info_queue = nullptr;
+	if (SUCCEEDED(device12->QueryInterface(&info_queue)))
+	{
+		CheckHR(info_queue->SetBreakOnSeverity(D3D12_MESSAGE_SEVERITY_ERROR, TRUE));
+		CheckHR(info_queue->SetBreakOnSeverity(D3D12_MESSAGE_SEVERITY_WARNING, TRUE));
+
+		D3D12_INFO_QUEUE_FILTER filter = {};
+		D3D12_MESSAGE_ID id_list[] = {
+			D3D12_MESSAGE_ID_CREATEGRAPHICSPIPELINESTATE_DEPTHSTENCILVIEW_NOT_SET, // Benign.
+			D3D12_MESSAGE_ID_CREATEGRAPHICSPIPELINESTATE_RENDERTARGETVIEW_NOT_SET, // Benign.
+			D3D12_MESSAGE_ID_CREATEINPUTLAYOUT_TYPE_MISMATCH, // Benign.
+			D3D12_MESSAGE_ID_DRAW_EMPTY_SCISSOR_RECTANGLE, // Benign. Probably.
+			D3D12_MESSAGE_ID_INVALID_SUBRESOURCE_STATE,
+			D3D12_MESSAGE_ID_MAP_INVALID_NULLRANGE, // Benign.
+			D3D12_MESSAGE_ID_EXECUTECOMMANDLISTS_GPU_WRITTEN_READBACK_RESOURCE_MAPPED, // Benign.
+			D3D12_MESSAGE_ID_RESOURCE_BARRIER_BEFORE_AFTER_MISMATCH // Benign. Probably.
+		};
+		filter.DenyList.NumIDs = ARRAYSIZE(id_list);
+		filter.DenyList.pIDList = id_list;
+		info_queue->PushStorageFilter(&filter);
+
+		info_queue->Release();
+
+		// Used at Close time to report live objects.
+		CheckHR(device12->QueryInterface(&s_debug_device12));
+	}
+
+	// prevent DXGI from responding to Alt+Enter, unfortunately DXGI_MWA_NO_ALT_ENTER
+	// does not work so we disable all monitoring of window messages. However this
+	// may make it more difficult for DXGI to handle display mode changes.
+	hr = factory->MakeWindowAssociation(wnd, DXGI_MWA_NO_WINDOW_CHANGES);
+	if (FAILED(hr))
+		MessageBox(wnd, _T("Failed to associate the window"), _T("Dolphin Direct3D 12 backend"), MB_OK | MB_ICONERROR);
+
+	SAFE_RELEASE(factory);
+	SAFE_RELEASE(output);
+	SAFE_RELEASE(adapter)
+
+	CreateDescriptorHeaps();
+	CreateRootSignatures();
+
+	command_list_mgr = new D3DCommandListManager(
+		D3D12_COMMAND_LIST_TYPE_DIRECT,
+		device12,
+		command_queue,
+		gpu_descriptor_heaps,
+		ARRAYSIZE(gpu_descriptor_heaps),
+		default_root_signature
+		);
+
+	command_list_mgr->GetCommandList(&current_command_list);
+	command_list_mgr->SetInitialCommandListState();
+
+	for (UINT i = 0; i < SWAP_CHAIN_BUFFER_COUNT; i++)
+	{
+		ID3D12Resource* buf12;
+		hr = s_swapchain->GetBuffer(i, IID_PPV_ARGS(&buf12));
+
+		s_backbuf[i] = new D3DTexture2D(buf12,
+			D3D11_BIND_RENDER_TARGET,
+			DXGI_FORMAT_UNKNOWN,
+			DXGI_FORMAT_UNKNOWN,
+			DXGI_FORMAT_UNKNOWN,
+			false,
+			D3D12_RESOURCE_STATE_PRESENT // Swap Chain back buffers start out in D3D12_RESOURCE_STATE_PRESENT.
+			);
+
+		CHECK(s_backbuf != nullptr, "Create back buffer texture");
```

Comment:
New won't return null, so no point checking the result here. Not sure if that was the intended condition, but checking if the array itself is not null will never fail either.